components/script/dom/globalscope.rs

```diff
@@ -2848,6 +2849,28 @@ impl GlobalScope {
         is_js_evaluation_allowed == CheckResult::Allowed
     }
 
+    pub(crate) fn should_navigation_request_be_blocked(&self, url: &ServoUrl) -> bool {
+        let Some(csp_list) = self.get_csp_list() else {
+            return false;
+        };
+        let request = Request {
+            url: url.clone().into_url(),
+            origin: url.origin().into_url_origin(),
+            redirect_count: 0,
```

Comment:
Can we have a TODO to populate this field correctly?

Reply:
Done. Not sure if `javascript:` URLs can redirect though. Isn't immediately obvious from the spec at least to me.